dftimewolf/lib/collectors/azure_logging.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+"""Reads logs from an Azure subscription."""
+import json
+import tempfile
+from typing import Optional
+
+from azure import identity as az_identity
+from azure.mgmt import monitor as az_monitor
+from azure.core import exceptions as az_exceptions
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class AzureLogsCollector(module.BaseModule):
+  """Collector for Azure Activity logs."""
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str]=None,
+               critical: bool=False) -> None:
+    """Initializes an Azure logs collector."""
+    super(AzureLogsCollector, self).__init__(
+        state, name=name, critical=critical)
+    self._filter_expression = ''
+    self._subscription_id = ''
+
+  # pylint: disable=arguments-differ
+  def SetUp(self, subscription_id: str, filter_expression: str) -> None:
+    """Sets up an Azure logs collector.
+
+    Args:
+      subscription_id (str): name of the subscription_id to fetch logs from.
+      filter_expression (str): Azure logs filter expression.
+    """
+    self._subscription_id = subscription_id
+    self._filter_expression = filter_expression
+
+  def Process(self) -> None:
+    """Copies logs from an Azure subscription."""
+
+    output_file = tempfile.NamedTemporaryFile(
+        mode='w', delete=False, encoding='utf-8', suffix='.jsonl')
+    output_path = output_file.name
+    self.logger.info('Downloading logs to {0:s}'.format(output_path))
+
+    credentials = az_identity.DefaultAzureCredential()
+    monitoring_client = az_monitor.MonitorManagementClient(
+        credentials, self._subscription_id)
+    activity_logs_client = monitoring_client.activity_logs
+
+    try:
+      results = activity_logs_client.list(filter=self._filter_expression)
+
+      while True:
```

Comment:
What guarantee do we have this won't infinite loop? `StopIteration` *MUST* be thrown at some point?

Reply:
`StopIteration` will be raised when there are no more log entries to be returned for the query (max time range is 90 days). Putting a limit on the loop would limit the number of log entries that could be collected.